draft-ietf-httpbis-digest-headers.md

```diff
@@ -318,6 +332,79 @@ Want-Digest: sha-256
 Want-Digest: sha-512;q=0.3, sha-256;q=1, unixsum;q=0
 ~~~
 
+
+# The Content-Digest Field {#content-digest}
+
+The `Content-Digest` field contains a comma-separated list of one or more content digest
+values.
+A content digest value is computed by applying a digest-algorithm to the actual message content
+(see Section 6.4 of {{SEMANTICS}}).
+It can be used in both requests and responses.
+
+~~~ abnf
+   Content-Digest = 1#content-digest
+   content-digest = digest-algorithm "="
+                    <encoded digest output>
+~~~
+
+For example:
+
+~~~ http-message
+Content-Digest: id-sha-512=WZDPaVn/7XgHaAy8pmojAkGWoRx2UFChF41A2svX+TaPm
+                           AbwAgBWnrIiYllu7BNNyealdVLvRwE\nmTHWXvJwew==
+~~~
+
+A `Content-Digest` field MAY contain multiple content-digest values,
+similarly to `Digest` (see {{digest}})
+
+~~~ http-message
+Content-Digest: sha-256=4REjxQ4yrqUVicfSKYNO/cF9zNj5ANbzgDZt3/h3Qxo=,
+                id-sha-256=X48E9qOokqqrvdts8nOJRJN3OWDUoyWxBf7kbu9DBPE=
+~~~
+
+A recipient MAY ignore any or all of the content-digests in a Content-Digest
+field. This allows the recipient to choose which digest-algorithm(s) to use for
+validation instead of verifying every received representation-data-digest.
+
+A sender MAY send a content-digest using a digest-algorithm without
+knowing whether the recipient supports the digest-algorithm, or even knowing
+that the recipient will ignore it.
+
+`Content-Digest` can be sent in a trailer section.
+In this case,
+`Content-Digest` MAY be merged in to the header section (See Section 6.5.1 of {{SEMANTICS}}).
+
+When an incremental digest-algorithm
+is used, the sender and the receiver can dynamically compute the digest value
+while streaming the content.
+
+# The Want-Content-Digest Field {#want-content-digest}
+
+The `Want-Content-Digest` field indicates the sender's desire to receive a content
+digest on messages associated with the request URI and representation metadata.
+It can be used in both requests and responses.
+
+~~~
+   Want-Content-Digest = 1#want-digest-value
```

Comment:
Given how similar Want-Digest and Want-Content-Digest are, and that they share the `want-digest-value` we can probably editorialize this a bit. But in order not to block this PR, we can do that work in a follow up.

Reply:
filed https://github.com/httpwg/http-extensions/issues/1632